resource-aggregate/events/resourceStateSnapshotTaken.go

```diff
@@ -80,8 +80,42 @@ func (e *ResourceStateSnapshotTaken) CheckInitialized() bool {
 		e.GetEventMetadata() != nil
 }
 
-func (e *ResourceStateSnapshotTaken) HandleEventResourceCreatePending(ctx context.Context, createPending *ResourceCreatePending) error {
+type resourceValidUntilValidator interface {
+	ValidUntilTime() time.Time
+	IsValid(now time.Time) bool
+	GetEventMetadata() *EventMetadata
+	GetAuditContext() *commands.AuditContext
+	GetResourceId() *commands.ResourceId
+}
+
+func (e *ResourceStateSnapshotTaken) processValidUntil(v resourceValidUntilValidator, isCmd bool, now time.Time) (bool, error) {
+	if !v.IsValid(now) {
```

Comment:
Validation doesn't belong to aggregate.

Reply:
It's bellows here because events can be expired during processing eventstore.